Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TeamsController.cs

```diff
@@ -716,8 +730,8 @@ private async Task<List<ShiftsIntegResponse>> ProcessSwapShiftRequestApprovalAsy
                 var newShiftSecond = JsonConvert.DeserializeObject<Shift>(postedShifts.Last().Body.ToString());
 
                 // Step 1 - Create the Kronos Unique ID.
-                var kronosUniqueIdFirst = this.utility.CreateUniqueId(newShiftFirst);
-                var kronosUniqueIdSecond = this.utility.CreateUniqueId(newShiftSecond);
+                var kronosUniqueIdFirst = this.utility.CreateUniqueId(newShiftFirst, timeZone);
+                var kronosUniqueIdSecond = this.utility.CreateUniqueId(newShiftSecond, timeZone);
```

Comment:
What happens if there are no postedShifts?

Reply:
Then it is not a valid shift swap from Teams - they are part of the contract - if a shift swap is approved then teams supplies a payload with 2 deleted shifts being the old shifts in the swap and 2 created shifts being the swapped shifts